reporter/src/main/kotlin/reporters/NoticeByPackageReporter.kt

```diff
@@ -54,7 +54,7 @@ class NoticeByPackageReporter : AbstractNoticeReporter() {
 
 class NoticeByPackageProcessor(input: ReporterInput) : AbstractNoticeReporter.NoticeProcessor(input) {
     companion object {
-        private val DEFAULT_ARCHIVE_DIR by lazy { getOrtDataDirectory().resolve("scanner/archive") }
+        private val DEFAULT_ARCHIVE_DIR by lazy { getOrtDataDirectory().resolve("archive") }
```

Comment:
This is a potentially breaking change for users that already have a scanner archive and should be mentioned in the commit message.
Maybe it would be better to put this change and the one for the scan results into separate commits. Also, in these cases it would make sense to keep the tool name in the folder name, because these files are created by the scanner. I understand the reasons for merging the HTTP caches in the commit message, but it doesn't explain why it is better to not have the scanner subdirectory in the data dir. Except "It simplifies the configuration of build cache directories on CI", but that does not apply here, because caching `datadir/archive` and `datadir/results` is not simpler than caching `datadir/scanner`.
If you are strongly against having the subdirectory, what about calling those `scan-archive` and `scan-results`?

These paths also have to be updated in the `reference.conf`.

Reply:
> I understand the reasons for merging the HTTP caches in the commit message, but it doesn't explain why it is better to not have the scanner subdirectory in the data dir.

I've split out the less controversial merging the HTTP caches into a separate PR: https://github.com/oss-review-toolkit/ort/pull/2569